### PR TITLE
filter (ticdc): add a lock to protect the parser from being called concurrently.

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/9571

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: scripts/check-diff-line-width.sh [hunk 1]
```diff
@@ -26,6 +26,7 @@ ERROR_THRESHOLD=140
 git --no-pager diff $BASE_HASH -U0 -- cdc pkg cmd \
 	-- ':(exclude)*_gen.go' \
 	-- ':(exclude)*_gen_test.go' \
+	-- ':(exclude)*_test.go' \
```
Comment (CharlesCheung96) [inline scripts/check-diff-line-width.sh:29]: It is better to use raw strings in the test and delete this line.

**Review by 3AceShowHand — APPROVED** [review]

File: scripts/check-diff-line-width.sh (hunk 1 above)
Comment (asddongmen) [reply scripts/check-diff-line-width.sh]: ok

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**